5.0/en/0x14-V6-Cryptography.md

```diff
@@ -32,23 +49,21 @@ Although this section is not easily penetration tested, developers should consid
 | **6.2.4** | Verify that the application is designed with crypto agility such that random number, encryption or hashing algorithms, key lengths, rounds, ciphers or modes can be reconfigured, upgraded, or swapped at any time, to protect against cryptographic breaks. Similarly, it must also be possible to replace keys and passwords and re-encrypt data. This should allow for seamless upgrades to post-quantum cryptography (PQC), once PQC standards are fully established. | | ✓ | ✓ | 320 |
-| **6.2.5** | [MODIFIED] Verify that known insecure block modes (i.e. ECB, etc.), padding modes (i.e. PKCS#1 v1.5, etc.), ciphers with small block sizes (i.e. Triple-DES, Blowfish, etc.), and weak hashing algorithms (i.e. MD5, SHA1, etc.) are not used. | | ✓ | ✓ | 326 |
-| **6.2.6** | [DELETED, DUPLICATE OF 6.5.3]| | | ✓ | 326 |
-| **6.2.7** | Verify that encrypted data is authenticated via signatures, authenticated cipher modes, or HMAC to ensure that ciphertext is not altered by an unauthorized party. | | | ✓ | 326 |
+| **6.2.5** | [SPLIT TO 6.5.1, 6.5.2, 6.6.3]. | | | | |
+| **6.2.6** | [MOVED TO 6.5.3]| | | | |
+| **6.2.7** | [MOVED TO 6.5.4]| | | | |
 | **6.2.8** | Verify that all cryptographic operations are constant-time, with no 'short-circuit' operations in comparisons, calculations, or returns, to avoid leaking information. | | | ✓ | 385 |
 | **6.2.9** | [ADDED] All cryptographic primitives MUST utilize a minimum of 128-bits of security, with exceptions only made for equipment or applications approaching end of life, where the requirement is at least 112-bits of security for all cryptography. | ✓ | ✓ | ✓ | 311 |
 
-
 ## V6.3 Random Values
 
 Cryptographically secure Pseudo-random Number Generation (CSPRNG) is incredibly difficult to get right. Generally, good sources of entropy within a system will be quickly depleted if over-used, but sources with less randomness can lead to predictable keys and secrets.
 
 | # | Description | L1 | L2 | L3 | CWE |
 | :---: | :--- | :---: | :---: | :---: | :---: |
-| **6.3.1** | [MODIFIED] All random numbers and strings intended to be non-guessable must be generated using a CSPRNG. | ✓ | ✓ | ✓ | 338 |
-| **6.3.2** | 6.3.2 [MODIFIED] Verify that GUIDs are created with an implementation of the GUID v4 or v7 algorithms which utilizes a CSPRNG.                                               |     |  ✓  |  ✓  | 338 |
-| **6.3.3** | Random number generation must work properly under heavy system load, or the system must degrade gracefully. | ✓ | ✓ | ✓ | 338 |
-
+| **6.3.1** | [GRAMMAR, LEVEL L2 > L1] Verify that all random numbers and strings which are intended to be non-guessable must be generated using a cryptographically-secure pseudo-random number generator (CSPRNG). | ✓ | ✓ | ✓ | 338 |
+| **6.3.2** | [MODIFIED] Verify that GUIDs are created with an implementation of the GUID v4 or v7 algorithms which utilizes a cryptographically-secure pseudo-random number generator (CSPRNG). | | ✓ | ✓ | 338 |
```

Comment:
One question about the use of UUIDs over GUIDs as per RC 4122, I'm easy and if it fits into the general style and approach, happy to adapt UUIDs